sources/booster.move

```diff
@@ -0,0 +1,175 @@
+    public entry fun claim_refund_boosted(
+        pub: &Publisher,
+        pool: &mut RefundPool,
+        affected_address: address,
+        new_address: address,
+        signature: vector<u8>,
+        ctx: &mut TxContext,
+    ) {
+        refund::assert_publisher(pub);
+        assert!(table::contains(unclaimed(pool), affected_address), EInvalidAddress); // TODO: get from assert function
+        // refund::assert_claim_phase(pool);
+        
+        // Reconstruct message
+        let pubkey = address_to_bytes(affected_address); // TODO: (Placeholder) this should be the pubkey converted in bytes
+        let affected_address_bytes = address_to_bytes(affected_address);
+        let new_address_bytes = address_to_bytes(new_address);
+
+        let msg = vector::empty();
+        vector::append(&mut msg, affected_address_bytes);
+        vector::append(&mut msg, new_address_bytes);
+        vector::append(&mut msg, object::id_to_bytes(&object::id(pool)));
+
+        assert!(
+            ed25519::ed25519_verify(&signature, &pubkey, &msg),
+            EIncorrectSignature
+        );
+        let refund_amount = *table::borrow(unclaimed(pool), affected_address);
+
+        // Base Refund
+        claim_refund_(pool, new_address, ctx);
+
+        // Booster Refund
+        let boost = div(refund_amount, 2);
+
+        let total_boosted = total_boosted_mut(accounting_mut(pool));
+        *total_boosted = *total_boosted + boost;
+
+        let booster = booster_pool_mut(pool);
+        let funds = balance::split(funds_mut(booster), boost);
+
+        transfer::public_transfer(coin::from_balance(funds, ctx), new_address);
+    }
+
+    // === Phase 4: Reclaim Fund ===
+
+    // TODO: Encapsulate logic in shared function with refund::reclaim_fund
+    public entry fun reclaim_fund(
+        pool: &mut RefundPool,
+        ctx: &mut TxContext,
+    ) {
+        refund::assert_reclaim_phase(pool);
+        let total_raised = total_raised_for_boost(accounting(pool));
+
+        let booster = booster_pool_mut(pool);
+
+        let funders = funders_mut(booster);
+        assert!(table::contains(funders, sender(ctx)), 0); // TODO: err code
+        let funding_amount = table::remove(funders, sender(ctx));
+
+        let is_last = table::is_empty(funders);
+        let funds = funds_mut(booster);
+        
+        let reclaim_amount = if (is_last) {
+            balance::value(funds)
+        } else {
+            let leftovers = balance::value(funds);
+
+            // ReclaimAmount = Leftovers * % Share <=>
+            // ReclaimAmount = Leftovers * FundingAmount/TotalRaised
+            // 
+            // We first upscale then downscale
+            div(
+                mul(leftovers, funding_amount),
+                total_raised
+            )
+        };
+
+        let reclaim_funds = coin::from_balance(balance::split(funds, reclaim_amount), ctx);
+        transfer::public_transfer(reclaim_funds, sender(ctx));
+    }
```

Comment:
Also useful to have an API for deleting `RefundPool` once `funds`/`funders` are empty

Reply:
Added